muzlag/internal/cmd/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/Drozd0f/gobots/muzlag/internal/cmd/bot"
+)
+
+func main() {
+	session := bot.NewBot()
+
+	sc := make(chan os.Signal, 1)
+	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
+	<-sc
+
+	if err := session.Close(); err != nil {
```

Comment:
How about to manage thing is the same place?

For example, if you are creating new Bot entity
```go
    bot := bot.New()
```

And you can use one of two approaches:
1. Bot implements `Run() error` and `Close() error`, so you can run bot, and close it if you got notification about Interrupt signal
2. Bot implements `Run(ctx context.Context) error`, and use `signal.NotifyContext`, so your bot could understand when it needs to be stopped (ctx.Done), and will handle graceful shutdown